tests/test_engine.py

```diff
@@ -52,6 +52,18 @@ def test_get_lookup_url(self):
         assert final_uri_with_params == '1/2/?extra_param=3'
         SampleResourceModel._lookup_urls = []
 
```

Comment:
@jbane @gaconnet thanks for the suggestion, explanation, and example. I have updated the test to use OrderedDict.